File: aspnet-in-practice/06/CS/CustomControls.ControlBuilders/Parts/Box.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;

namespace CustomControls.ControlBuilders
{
	// if we inherit from Zone, we can encapsulate other elements as in the main control
	public class Box: Zone
	{
		protected override void Render(HtmlTextWriter writer)
		{
			if (!string.IsNullOrEmpty(Title))
			{
				writer.WriteBeginTag("h2");
				writer.Write(">");
				writer.Write(Title);
				writer.WriteEndTag("h2");
			}

			base.Render(writer);
		}
	}
}

File: aspnet-in-practice/06/VB/CustomControls.Web/CustomControls.ControlBuilders/Parts/Box.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Web.UI

' if we inherit from Zone, we can encapsulate other elements as in the main control
Public Class Box
	Inherits Zone
	Protected Overrides Sub Render(ByVal writer As HtmlTextWriter)
		If Not String.IsNullOrEmpty(Title) Then
			writer.WriteBeginTag("h2")
			writer.Write(">")
			writer.Write(Title)
			writer.WriteEndTag("h2")
		End If

		MyBase.Render(writer)
	End Sub
End Class